internal/cmd/evergreen/main.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2022 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/evergreen-ci/shrub"
+)
+
+const atlascli = "atlascli"
+const mongocli = "mongocli"
+
+func buildDependency(toolName, os, serverVersion, repo string) shrub.TaskDependency {
+	newOs := map[string]string{
+		"centos7":      "rhel70",
+		"centos8":      "rhel80",
+		"amazonlinux2": "amazon2",
+		"ubuntu18.04":  "ubuntu1804",
+		"ubuntu20.04":  "ubuntu2004",
+		"debian9":      "debian92",
+		"debian10":     "debian10",
+	}
+
+	newRepo := map[string]string{
+		"org": "org",
+		"ent": "enterprise",
+	}
+
+	return shrub.TaskDependency{
+		Name:    fmt.Sprintf("push_%v_%v_%v_stable", toolName, newOs[os], newRepo[repo]),
+		Variant: fmt.Sprintf("release_%v_publish_%v", toolName, strings.ReplaceAll(serverVersion, ".", "")),
+	}
+}
+
+func generateRepoTasks(toolName string) *shrub.Configuration {
+	serverVersions := []string{"4.2", "4.4", "5.0"}
+	oses := []string{"amazonlinux2", "centos7", "centos8", "debian9", "debian10", "ubuntu18.04", "ubuntu20.04"}
+	repos := []string{"org", "ent"}
+
+	c := &shrub.Configuration{}
+
+	for _, serverVersion := range serverVersions {
+		v := &shrub.Variant{
+			BuildName:        fmt.Sprintf("test_repo_%v_%v", toolName, serverVersion),
+			BuildDisplayName: fmt.Sprintf("Test %v on repo %v", toolName, serverVersion),
+			DistroRunOn:      []string{"ubuntu1804-small"},
+		}
+
+		pkg := "mongodb-atlas-cli"
+		entrypoint := "atlas"
+		if toolName == mongocli {
+			pkg = mongocli
+			entrypoint = mongocli
+		}
+
+		for _, os := range oses {
+			for _, repo := range repos {
+				t := &shrub.Task{
+					Name: fmt.Sprintf("test_repo_%v_%v_%v_%v", toolName, os, repo, serverVersion),
+				}
+				t = t.Stepback(false)
+				t = t.GitTagOnly(true)
+				t = t.Dependency(buildDependency(toolName, os, serverVersion, repo))
```

Comment:
[q] would this be the same as
```suggestion
				t.Stepback(false)
				.GitTagOnly(true)
				.Dependency(buildDependency(toolName, os, serverVersion, repo))
```